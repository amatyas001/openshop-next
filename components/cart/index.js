import { CartContent } from './CartContent/CartContent';
import { CartAmount } from './CartAmount/CartAmount';
import { CartButton } from './CartButton/CartButton';
import { CartItem } from './CartItem/CartItem';
import { CartPanel } from './CartPanel/CartPanel';

export { CartContent, CartAmount, CartButton, CartItem, CartPanel };
