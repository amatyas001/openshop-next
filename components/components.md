Elements are composed by [Chakra-UI](https://chakra-ui.com) components and preserve the library features so [Style Props](https://chakra-ui.com/style-props) can be passed to them and accessibility remains optimized.

> _Style Props are a way to alter the style of a component by simply passing props to it. It helps to save time by providing helpful shorthand ways to style components._

For more information visit the [Chakra UI](https://chakra-ui.com) documentation.

Components are divided into two main category: _wrappers_ and _single responsibility_ elements. Wrapper components does not contain business logic, however there are special cases like _handling lifted states to keep sensitive data locally_ and not to populate them to the Redux store which is persisted on the machine.

**State Dependencies** and **Wrapped Components** are documented in each description to give brief information about the context of posible interactions of the actual component. _Examples_ prepared with proper state and may render extra controls to test different cases with ease.
