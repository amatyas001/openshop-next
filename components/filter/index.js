export * from './FilterButton/FilterButton';
export * from './FilterColor/FilterColor';
export * from './FilterPrice/FilterPrice';
export * from './FilterName/FilterName';
export * from './FilterPanel/FilterPanel';
export * from './FilterReset/FilterReset';
