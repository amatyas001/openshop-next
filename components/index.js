export * from './shared';
export * from './cart';
export * from './payment';
