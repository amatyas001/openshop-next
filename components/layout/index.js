export * from './Content/Content';
export * from './Footer/Footer';
export * from './ProductsTopBar/ProductsTopBar';
export * from './Navbar/Navbar';
export * from './ScrollTop/ScrollTop';
