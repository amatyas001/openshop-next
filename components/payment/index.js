export * from './PaymentCancelled/PaymentCancelled';
export * from './PaymentConfirm/PaymentConfirm';
export * from './PaymentConfirmCard/PaymentConfirmCard';
export * from './PaymentConfirmControls/PaymentConfirmControls';
export * from './PaymentConfirmDetails/PaymentConfirmDetails';
export * from './PaymentError/PaymentError';
export * from './PaymentForm/PaymentForm';
export * from './PaymentFormControls/PaymentFormControls';
export * from './PaymentProgress/PaymentProgress';
export * from './PaymentReview/PaymentReview';
export * from './PaymentSuccess/PaymentSuccess';
