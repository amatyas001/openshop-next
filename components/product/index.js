import { ProductCard } from './ProductCard/ProductCard';
import { ProductControls } from './ProductControls/ProductControls';
import { ProductControlsColor } from './ProductControlsColor/ProductControlsColor';
import { ProductControlsAmount } from './ProductControlsAmount/ProductControlsAmount';
import { ProductControlsButton } from './ProductControlsButton/ProductControlsButton';
import { ProductControlsSize } from './ProductControlsSize/ProductControlsSize';

export {
  ProductCard,
  ProductControls,
  ProductControlsColor,
  ProductControlsAmount,
  ProductControlsButton,
  ProductControlsSize,
};
