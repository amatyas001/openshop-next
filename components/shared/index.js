import { Button } from './Button/Button';
import { ButtonIcon } from './ButtonIcon/ButtonIcon';
import { Field } from './Field/Field';
import { Image } from './Image/Image';
import { Panel } from './Panel/Panel';
import { Spinner } from './Spinner/Spinner';

export { Button, ButtonIcon, Field, Image, Panel, Spinner };
