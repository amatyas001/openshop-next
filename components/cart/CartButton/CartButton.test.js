import { create, act } from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { panelToggle } from '@app/lib/redux/actions';
import { mockProductWithBuyAmount } from '@app/mocks';
import { CartButton } from '@app/components';

let tree;
let store;
const mockStore = configureStore([]);

describe('<CartButton />', () => {
  describe('on empty cart', () => {
    beforeAll(() => {
      store = mockStore({
        cart: [],
        panel: {
          cart: false,
        },
      });
      act(() => {
        tree = create(
          <Provider store={store}>
            <CartButton />
          </Provider>
        );
      });
    });

    it('should render without props', () => {
      expect.assertions(1);
      expect(tree.toJSON()).toMatchSnapshot();
    });

    it('should toggle panel collapse', () => {
      expect.assertions(1);
      act(() => {
        tree.root.findByType('button').props.onClick();
      });
      expect(store.getActions()).toEqual([
        panelToggle('cart', !store.getState().panel.cart),
      ]);
    });
  });

  describe('on not empty cart', () => {
    const amount = 1;

    beforeAll(() => {
      store = mockStore({
        cart: mockProductWithBuyAmount(1, amount),
        panel: { cart: false },
      });
      act(() => {
        tree = create(
          <Provider store={store}>
            <CartButton />
          </Provider>
        );
      });
    });

    it('should render without props', () => {
      expect.assertions(1);
      expect(tree.toJSON()).toMatchSnapshot();
    });

    it('should toggle panel collapse', () => {
      expect.assertions(1);
      act(() => {
        tree.root.findByType('button').props.onClick();
      });
      expect(store.getActions()).toEqual([
        panelToggle('cart', !store.getState().panel.cart),
      ]);
    });

    it('should display amount of cart items', () => {
      expect.assertions(1);
      expect(tree.root.findByType('button').props.children).toContain(
        `${amount} item(s)`
      );
    });
  });
});
