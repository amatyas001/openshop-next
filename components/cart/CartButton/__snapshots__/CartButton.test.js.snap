// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartButton /> on empty cart should render without props 1`] = `
<button
  aria-label="cart-toggler"
  className="css-1vus3us"
  href="#void"
  id="cart-toggler"
  onClick={[Function]}
  onKeyDown={[Function]}
  tabIndex="0"
  type="button"
>
  <svg
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
        "marginBottom": "-5px",
        "marginRight": "5px",
      }
    }
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zM1 2v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.14 0-.25-.11-.25-.25l.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.58-6.49c.08-.14.12-.31.12-.48 0-.55-.45-1-1-1H5.21l-.94-2H1zm16 16c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2z"
    />
  </svg>
  0 item(s)
</button>
`;

exports[`<CartButton /> on not empty cart should render without props 1`] = `
<button
  aria-label="cart-toggler"
  className="css-1vus3us"
  href="#void"
  id="cart-toggler"
  onClick={[Function]}
  onKeyDown={[Function]}
  tabIndex="0"
  type="button"
>
  <svg
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
        "marginBottom": "-5px",
        "marginRight": "5px",
      }
    }
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zM1 2v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.14 0-.25-.11-.25-.25l.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.58-6.49c.08-.14.12-.31.12-.48 0-.55-.45-1-1-1H5.21l-.94-2H1zm16 16c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2z"
    />
  </svg>
  1 item(s)
</button>
`;
